foundry/run_node.rs

```diff
@@ -269,6 +270,9 @@ pub fn run_node(matches: &ArgMatches<'_>) -> Result<(), String> {
             let peer_db = PeerDb::new(c.get_kvdb());
             let service = network_start(network_id, timer_loop, &network_config, Arc::clone(&routing_table), peer_db)?;
 
+            let auto_self_nominate = AutoSelfNomination::new(c);
```

Comment:
CodeChain must run the auto-self-nominate thread only when a user enabled it. We need a commandline flag for it.